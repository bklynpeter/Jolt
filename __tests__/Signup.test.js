import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "React";
import Signup from "../client/Components/Signup";
import { unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from "react-test-renderer";

/**
 * @jest-environment jsdom
 */

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

test("Baseline", () => {
  expect(true).not.toBe(false);
});

test('Renders something with "username"', () => {
  render(<Signup />, container);
  const username = screen.queryByText("username");
  expect(username).toBeTruthy();
});

test('Renders something with "password"', () => {
  render(<Signup />, container);
  const password = screen.queryByText("Password");
  expect(password).toBeTruthy();
});

test('Does NOT render something with "blah"', () => {
  render(<Signup />, container);
  const blah = screen.queryByText("blah");
  expect(blah).not.toBeTruthy();
});

test('Renders something with test-id "PersonIcon"', () => {
  render(<Signup />, container);
  const PersonIcon = screen.queryByTestId("PersonIcon");
  expect(PersonIcon).toBeTruthy();
});

test('Renders something with test-id "CheckBox"', () => {
  render(<Signup />, container);
  const CheckBox = screen.queryByTestId("CheckBox");
  expect(CheckBox).not.toBeTruthy();
});

test('Does NOT render something with test-id "noId"', () => {
  render(<Signup />, container);
  const noId = screen.queryByTestId("noId");
  expect(noId).not.toBeTruthy();
});

test("Renders at least one button", () => {
  render(<Signup />, container);
  const button = screen.queryByRole("button");
  expect(button).toBeTruthy();
});

test("Does NOT render at least one table", () => {
  render(<Signup />, container);
  const table = screen.queryByRole("table");
  expect(table).not.toBeTruthy();
});

test('Does NOT render something called the "sign in" button', () => {
  render(<Signup />, container);
  const button = screen.queryByTestId("signInButton");
  expect(button).not.toBeTruthy();
});

test('The "Sign Up" button is actually a button', () => {
  render(<Signup />, container);
  const signInButton = screen.getByTestId("signUpButton");
  expect(signInButton).toBeInstanceOf(HTMLButtonElement);
});

test('Does NOT find the "Sign In" button', () => {
  render(<Signup />, container);
  const signInButtonArray = screen.queryAllByTestId("signInButton");
  expect(signInButtonArray).toEqual([]);
});

test("Should identify if the spy function was called or not", () => {
  const spyFunction = jest.fn();
  spyFunction();
  expect(spyFunction).toHaveBeenCalled();
});

test("Renders Signup correctly per Snapshot", () => {
  const signupTest = renderer.create(<Signup />).toJSON();
  expect(signupTest).toMatchSnapshot();
});
