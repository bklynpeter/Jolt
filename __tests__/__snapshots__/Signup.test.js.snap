// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Signup correctly per Snapshot 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
>
  <div
    className="MuiBox-root css-binzgt"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1999axt-MuiAvatar-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
        data-testid="PersonIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Sign up
    </h1>
    <form
      className="MuiBox-root css-178yklu"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="username"
              id="username-label"
            >
              username
              <span
                aria-hidden={true}
                className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="username"
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="username"
                name="username"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-1ftyaf0"
                >
                  <span>
                    username
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="password"
              id="password-label"
            >
              Password
              <span
                aria-hidden={true}
                className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="new-password"
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="password"
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="password"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-1ftyaf0"
                >
                  <span>
                    Password
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
        data-testid="signUpButton"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        Sign Up
      </button>
    </form>
  </div>
</main>
`;
